tests/templates/test_subroutines/test_fable.py

```diff
@@ -391,89 +379,35 @@ def circuit_lightning():
     @pytest.mark.parametrize(
         ("input", "wires"),
         [
-            (
-                np.random.random((1, 2)),
-                3,
-            ),
-            (
-                np.random.random((1, 1)),
-                3,
-            ),
-            (
-                np.random.random((2, 1)),
-                3,
-            ),
-            (
-                np.random.random((3, 2)),
-                5,
-            ),
-            (
-                np.random.random((4, 2)),
-                5,
-            ),
-            (
-                np.random.random((2, 3)),
-                5,
-            ),
-            (
-                np.random.random((2, 4)),
-                5,
-            ),
-            (
-                np.random.random((3, 4)),
-                5,
-            ),
-            (
-                np.random.random((4, 3)),
-                5,
-            ),
-            (
-                np.random.random((3, 5)),
-                7,
-            ),
-            (
-                np.random.random((3, 6)),
-                7,
-            ),
-            (
-                np.random.random((3, 7)),
-                7,
-            ),
-            (
-                np.random.random((4, 5)),
-                7,
-            ),
-            (
-                np.random.random((5, 5)),
-                7,
-            ),
-            (
-                np.random.random((6, 5)),
-                7,
-            ),
-            (
-                np.random.random((2, 9)),
-                9,
-            ),
-            (
-                np.random.random((9, 3)),
-                9,
-            ),
+            (np.random.random((1, 2)), 3),
+            (np.random.random((1, 1)), 3),
+            (np.random.random((2, 1)), 3),
+            (np.random.random((3, 2)), 5),
+            (np.random.random((4, 2)), 5),
+            (np.random.random((2, 3)), 5),
+            (np.random.random((2, 4)), 5),
+            (np.random.random((3, 4)), 5),
+            (np.random.random((4, 3)), 5),
+            (np.random.random((3, 5)), 7),
+            (np.random.random((3, 6)), 7),
+            (np.random.random((3, 7)), 7),
+            (np.random.random((4, 5)), 7),
+            (np.random.random((5, 5)), 7),
+            (np.random.random((6, 5)), 7),
         ],
     )
     def test_variety_of_matrix_shapes(self, input, wires):
-        """Test that FABLE runs without error for variety of cases."""
+        """Test that FABLE runs without error for a variety of input shapes."""
         dev = qml.device("default.qubit")
-        s = int(qml.math.ceil(qml.math.log2(max(len(input), len(input[0])))))
-        dimension = s**2
+        s = int(qml.math.ceil(qml.math.log2(max(*input.shape))))
+        dim = 2**s
```

Comment:
This was a bug: The dimension `s**2` does not make sense here, as far as I know.
Instead, it should be `2**s`.
This was not caught because this test was not really testing the output, but slicing into the output only to then check that the sliced matrix has the right shape :sweat_smile:

Reply:
Yeah that's a bug. I recall fixing it for the actual template itself but I guess it slipped this particular test. Thanks @dwierichs for catching this :)